testbed/testbed/receivers.go

```diff
@@ -187,71 +187,90 @@ func (jr *JaegerDataReceiver) ProtocolName() string {
 }
 
 // OTLPDataReceiver implements OTLP format receiver.
-type OTLPDataReceiver struct {
+type baseOTLPDataReceiver struct {
 	DataReceiverBase
+	protocol        string
 	traceReceiver   component.TraceReceiver
 	metricsReceiver component.MetricsReceiver
 	logReceiver     component.LogsReceiver
 }
 
-// Ensure OTLPDataReceiver implements DataReceiver.
-var _ DataReceiver = (*OTLPDataReceiver)(nil)
-
-const DefaultOTLPPort = 55680
-
-// NewOTLPDataReceiver creates a new OTLPDataReceiver that will listen on the specified port after Start
-// is called.
-func NewOTLPDataReceiver(port int) *OTLPDataReceiver {
-	return &OTLPDataReceiver{DataReceiverBase: DataReceiverBase{Port: port}}
-}
-
-func (or *OTLPDataReceiver) Start(tc consumer.TraceConsumer, mc consumer.MetricsConsumer, lc consumer.LogsConsumer) error {
+func (bor *baseOTLPDataReceiver) Start(tc consumer.TraceConsumer, mc consumer.MetricsConsumer, lc consumer.LogsConsumer) error {
 	factory := otlpreceiver.NewFactory()
 	cfg := factory.CreateDefaultConfig().(*otlpreceiver.Config)
-	cfg.SetName(or.ProtocolName())
-	cfg.GRPC.NetAddr = confignet.NetAddr{Endpoint: fmt.Sprintf("localhost:%d", or.Port), Transport: "tcp"}
-	cfg.HTTP = nil
+	cfg.SetName(bor.protocol)
+	if bor.protocol == "otlp" {
```

Comment:
I think this needs to be "grpc" or "otlp/grpc".

Reply:
the "protocol" a.k.a exporterType has to be the type of one of our exporters, and in this case the exporter type is "otlp" for OTLP over gRPC